### PR TITLE
SG_ADD refactor

### PR DESCRIPTION
**Pull request by gf712**
I applied the same refactoring to SG_ADD as in [here](https://github.com/shogun-toolbox/shogun/pull/4417).

### CODE REVIEW

**Comment by karlnapf** [comment]
let's do it :)

File: shogun/multiclass/MulticlassLogisticRegression.cpp
Comment (karlnapf) [inline shogun/multiclass/MulticlassLogisticRegression.cpp:47]: we should introduce an attribute for parameters like this one .... these are hyper-parameters, but not in the classical sense (as in that we want to tune them). But still free parameters ...
@vigsterkr @lisitsyn ?

**Review by karlnapf — APPROVED** [review]

**Comment by karlnapf** [comment]
ah nevermind

**Comment by karlnapf** [comment]
once this submodule update is merged, you will need to update the submodule commit hash in the main develop repo, so that it is included in the develop branch

**Comment by karlnapf** [comment]
Minor: it would be good to write more meaningful commit messages so that we can later easily see what has happened

**Comment by gf712** [comment]
> once this submodule update is merged, you will need to update the submodule commit hash in the main develop repo, so that it is included in the develop branch

OK, I will start a PR with that.
> Minor: it would be good to write more meaningful commit messages so that we can later easily see what has happened

OK! I can write something with more information with a reference to the PR to shogun. I can write that in the PR message?

**Comment by karlnapf** [comment]
> > once this submodule update is merged, you will need to update the submodule commit hash in the main develop repo, so that it is included in the develop branch
> 
> OK, I will start a PR with that.
Single commit PR

> 
> > Minor: it would be good to write more meaningful commit messages so that we can later easily see what has happened
> 
> OK! I can write something with more information with a reference to the PR to shogun. I can write that in the PR message?
in the commit hash update, I would just say: update to latest version. But in the commit hash within the gpl submodule, you write what you did with the PR

**Comment by gf712** [comment]
How can I edit the message of the existing commit?

Just did 
```
git commit --amend
git push origin --force
```
How can I push to here?

**Comment by karlnapf** [comment]
Yes that’s it
You can only harm yourself (loose changes) with force pushing as you don’t have push rights to the main repo, so pretty safe

**Comment by gf712** [comment]
So do I start a new PR with the amended commit message?

**Comment by karlnapf** [comment]
Nono just amend and force push. Your pr will be overwritten/updated

**Pull request merged**
